lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -794,37 +785,48 @@ protected function getRenameIndexSQL($oldIndexName, Index $index, $tableName)
         ];
     }
 
+    private function quoteSingleIdentifierAsStringLiteral(string $levelName): string
+    {
+        return $this->quoteStringLiteral(preg_replace('~^\[|\]$~s', '', $levelName));
```

Comment:
So this removes brackets. Please explain why, it's the first time I see brackets in SQL (is that specific to the T-SQL thing you mentioned earlier?)

Reply:
yes - `` ` `` is MySQL, `[` (and `]`) in T-SQL, just different SQL grammar

Reply:
Ok, this part looks super ugly and fragile, do we really need it? Why are there brackets in the name in the first place?

Reply:
Strictly needed, because name can already be escaped, because of DBAL escaping approach now which allows it

Reply:
You mean the DBAL adds those brackets somewhere?

Reply:
no in general - DBAL allows name to be specified in unquoted and quoted string form, related with https://github.com/doctrine/dbal/issues/4385, which we will definitely not solve in this PR